File: 2838-sum-of-matrix-after-queries/sum-of-matrix-after-queries.java
class Solution {
    public long matrixSumQueries(int n, int[][] q) {
        long totalSum = 0, rowCount = 0, colCount = 0;
        HashSet<Integer> updatedRows = new HashSet<>();
        HashSet<Integer> updatedCols = new HashSet<>();

        for (int indx = q.length - 1; indx >= 0; indx--) {
            var query = q[indx];
            int operationType = query[0];
            int index = query[1];
            int value = query[2];

            if (operationType == 0 && !updatedRows.contains(index)) {
                updatedRows.add(index);
                rowCount++;
                totalSum += (value * (n - colCount));
            } else if (operationType == 1 && !updatedCols.contains(index)) {
                updatedCols.add(index);
                colCount++;
                totalSum += (value * (n - rowCount));
            }
        }

        return totalSum;
    }
}

File: 2838-sum-of-matrix-after-queries/sum-of-matrix-after-queries.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

class Solution {
public:
    long matrixSumQueries(int n, std::vector<std::vector<int>>& queries) {
        long totalSum = 0, updatedRowCount = 0, updatedColCount = 0;
        std::unordered_set<int> updatedRows, updatedCols;

        for (int idx = queries.size() - 1; idx >= 0; idx--) {
            auto query = queries[idx];
            int operationType = query[0];
            int index = query[1];
            int value = query[2];

            if (operationType == 0 && updatedRows.find(index) == updatedRows.end()) {
                updatedRows.insert(index);
                updatedRowCount++;
                totalSum += static_cast<long>(value) * (n - updatedColCount);
            } else if (operationType == 1 && updatedCols.find(index) == updatedCols.end()) {
                updatedCols.insert(index);
                updatedColCount++;
                totalSum += static_cast<long>(value) * (n - updatedRowCount);
            }
        }

        return totalSum;
    }
};